tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -973,6 +977,21 @@ public void expectCompletion(long timeoutMillis, String errorMsg) throws Interru
       } // else, ok
     }
 
+    public boolean tryExpectCompletion(long timeoutMillis) throws InterruptedException {
+      while (timeoutMillis-- > 0) {
+        Optional<T> value = abq.peek();
+        if (value != null) {
+          if (!value.isDefined()) {
+            abq.poll();
+            return true;
+          }
+          return false;
+        }
+        Thread.sleep(1);
```

Comment:
I don't understand this question. This routine uses peek because in case the next received element is not a terminal indicator, that element should remain available for any of the other `next` calls that may want to verify the item.